src/Form/SettingsForm.php

```diff
@@ -210,21 +200,21 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#tree' => TRUE,
     ];
 
-    $options = array_filter(
-      [
-        'openid_connect' => $this->t('OpenId Connect'),
-        'samlauth' => $this->t('SAML Authentication'),
-      ],
-      [$this->moduleHandler, 'moduleExists'],
-      ARRAY_FILTER_USE_KEY
-    );
+    $options = [];
+    $providers = $this->userManager->getActiveOpenIdConnectProviders();
 
-    $form['include']['modules'] = [
+    foreach ($providers as $provider => $value) {
+      // Drupal config cannot contain ".", why I have a different notation for
+      // key and value.
+      $options[$this->escapeProviderId($provider)] = $value;
+    }
```

Comment:
We should rename the UserManager's `getActiveOpenIdConnectProviders` function to `getAllUserProviders`, say, since we should not only care about OpenID Connect providers. We currently use only OIDC, but writing generic code is often easier than writing code for specific use cases.

`UserManager::getAllUserProviders` can take care of escaping config keys, or we can add a `getAllUserProvidersAsConfigOptions` to be more explicit.